ci/derived-image-test.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# This script performs the following tasks:
+#
+# 1. Logs into the rhcos-devel service account to get the image push credential
+# so we can push to the registry.ci.openshift.org/rhcos-devel namespace.
+#
+# 2. Kicks off the build-test-qemu.sh script to build and test an RHCOS image
+# with Kola tests.
+#
+# 3. Pushes the resulting image to the image registry using the Prow build ID
+# as the tag since we need to have the image in this registry so our derived
+# image build test can use it.
+#
+# 3. Kicks off the derived OS image testing binary which pulls the newly built
+# image, builds a derived image from it, applies it to an underlying OpenShift
+# cluster node, verifies that it was successfully applied, then rolls the node
+# back.
+
+set -euo pipefail
+
+# Note: The oc binary will be injected by the Prow CI process. It
+# is not present in ci/Dockerfile.
+oc login https://api.ci.l2s4.p1.openshiftapps.com:6443 --token="$(cat /service-account-token/image-pusher-service-account-token)"
+oc registry login --registry=registry.ci.openshift.org --to="$SHARED_DIR/dockercfg.json"
+
+export COSA_DIR="/tmp/cosa"
+mkdir -p "$COSA_DIR"
+
+# Run the cosa build / test
+/src/ci/build-test-qemu.sh
+
+export REGISTRY_AUTH_FILE="$SHARED_DIR/dockercfg.json"
+
+# Ensure we're in the designated cosa directory so the push-container commands work
+cd "$COSA_DIR"
+
+# Tags with the cosa build ID / arch - unique to this specific build
```

Comment:
Will anything prune these tags by default?  From a quick look at openshift/release I see [image pruners](https://docs.openshift.com/container-platform/4.10/applications/pruning-objects.html) set up on the CI clusters but not the central CI registry?

Reply:
I'm not sure to be honest. It would be nice to push to the ephemeral CI registry first so one doesn't have to worry about pruning these tags.